src/tests/ReactUnusedPropsAndStateRuleTests.ts

```diff
@@ -614,4 +614,413 @@ describe('reactUnusedPropsAndStateRule', (): void => {
             }
         ]);
     });
+
+    describe('using SFCs', () => {
```

Comment:
These tests look great!

But - if you're going to add more, you might have an easier time using the new test format (#489). That can be done by adding a new tests folder as in #806 and doesn't necessitate deleting the existing tests in this file. #489 tracks the repository migrating all files that way.

Reply:
After playing around with this I ended up just porting the old tests in #826, and I'll rebase this PR on top of that so all of the tests in this PR are also written using the inbuilt `tslint` test framework.